uaclient/entitlements/repo.py

```diff
@@ -144,6 +151,43 @@ def operational_status(self):
             return status.ACTIVE, '%s is active' % self.title
         return status.INACTIVE, '%s is not configured' % self.title
 
+    def process_contract_deltas(
+            self, orig_access: 'Dict[str, Any]',
+            deltas: 'Dict[str, Any]') -> None:
+        """Process any contract access deltas for this entitlement.
+
+        :param orig_access: Dictionary containing the original
+            resourceEntitlement access details.
+        :param deltas: Dictionary which contains only the changed access keys
+        and values.
+        """
+        if not deltas:
+            return
+        super().process_contract_deltas(orig_access, deltas)
```

Comment:
It looks to me like this super-call could end up disabling the entire entitlement, in which case there isn't much point doing anything else. I don't have a particularly good idea of how to short-circuit here, but is it something we should consider?

Reply:
Dan, let me know what you think here. I've changed the operation to return True if it has already performed operations. repo.process_contract_deltas will exit early of True returned from the super call.